test/unit/sdam/server.test.ts

```diff
@@ -0,0 +1,162 @@
+/* eslint-disable @typescript-eslint/no-non-null-assertion */
```

Comment:
I didn't see anything in the unskipped spec tests, but should we add tests for stale generation errors to ensure that we do not handle errors from a stale generation?

Reply:
There is a set of unit tests covering it